### PR TITLE
[Hotfix] The MySQL Connector/J is incompatible with the Apache 2.0 license.

### PR DESCRIPTION
**Pull request by zhuangchong**
## What changes were proposed in this pull request

The binary package lib directory contains `mysql-connector-java-8.0.27.jar`

Remove this jar because MySQL Connector/J is not compatible with Apache 2.0 licenses.

image

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / no)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -240,6 +240,7 @@
                 <groupId>mysql</groupId>
                 <artifactId>mysql-connector-java</artifactId>
                 <version>${mysql.version}</version>
+                <scope>test</scope>
```
Comment (SbloodyS) [inline pom.xml:243]: We should add this to the docs to tell users downloading mysql jar locally.

Comment (zhuangchong) [reply pom.xml:243]: There are instructions in the deployment document (https://streampark.apache.org/docs/platform/platform-build#introduce-mysql-dependency-package). At the same time, the instructions will be checked when starting the project (https://github.com/apache/incubator-streampark/blob/f9421f6bbc162200cd9accdc25ac2bd43e78e889/streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/base/config/SpringProperties.java#L56-L69)

Comment (SbloodyS) [reply pom.xml:243]: I see. Thanks for clarification.

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**